### PR TITLE
fix: few things

### PR DESCRIPTION
**Pull request by vgorkavenko**
- `ValidatorState` and  `Validator` typings
- cast `str` to `int` in `_get_with_blockstamp`
- fix `_get_first_non_missed_slot`

### CODE REVIEW

File: src/providers/keys/client.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def _get_with_blockstamp(self, url: str, blockstamp: BlockStamp, params: Optiona
         """
         for i in range(self.RETRY_COUNT):
             data, meta = self._get(url, params)
-            if meta['meta']['elBlockSnapshot']['blockNumber'] >= blockstamp.block_number:
+            if meta['meta']['elBlockSnapshot']['blockNumber'] >= int(blockstamp.block_number):
```
Comment (F4ever) [inline src/providers/keys/client.py:33]: This is already int

File: src/modules/submodules/consensus.py [hunk 2]
```diff
@@ -70,7 +71,7 @@ def _get_member_info(self, blockstamp: BlockStamp) -> MemberInfo:
 
         # Defaults for dry mode
         current_ref_slot, deadline_slot = self._get_current_frame(blockstamp)
-        is_member, last_report_ref_slot, member_report_for_current_ref_slot = True, 0, b''
+        is_member, last_member_report_ref_slot, member_report_for_current_ref_slot = True, 0, b''
```
Comment (F4ever) [inline src/modules/submodules/consensus.py:74]: It changes nothing

File: src/providers/consensus/typings.py [hunk 3]
```diff
@@ -42,16 +42,16 @@ class ValidatorState:
     withdrawal_credentials: str
     effective_balance: str
     slashed: bool
-    activation_eligibility_epoch: bool
+    activation_eligibility_epoch: str
```
Comment (F4ever) [inline src/providers/consensus/typings.py:45]: :+1:

Comment (F4ever) [inline src/providers/consensus/typings.py:45]: :+1:

File: src/modules/submodules/consensus.py (hunk 2 above)
Comment (vgorkavenko) [reply src/modules/submodules/consensus.py:74]: in dry mode `last_report_ref_slot` var was not used in any way further. I think it should be initial value for `last_member_report_ref_slot` to init `MemberInfo`

File: src/modules/submodules/consensus.py [hunk 4]
```diff
@@ -101,7 +102,7 @@ def _get_member_info(self, blockstamp: BlockStamp) -> MemberInfo:
 
         return MemberInfo(
             is_member=is_member,
-            last_report_ref_slot=last_report_ref_slot,
+            last_report_ref_slot=last_member_report_ref_slot,
```
Comment (F4ever) [inline src/modules/submodules/consensus.py:105]:
```suggestion
            last_report_ref_slot=last_report_ref_slot,
```

Comment (F4ever) [inline src/modules/submodules/consensus.py:105]:
```suggestion
            last_report_ref_slot=last_report_ref_slot,
```

**Review by F4ever — APPROVED** [review]

**Pull request merged**